.github/pull_request_template.md

```diff
@@ -16,3 +16,4 @@ Please describe the tests that you ran to verify your changes. Provide instructi
 - [ ] My changes generate no new warnings
 - [ ] All submodules modifications have been approved and merged into dev prior to this one
 - [ ] All submodules I have modified point to their latest main branch
+- [ ] I have updated the documentation related to my modification (Statemachine flow charts, manuals, architecture documentation)
```

Comment:
Lets see how this goes. We may have to put these in separate bullets if they all do not get done.